File: www/KeyboardPlugin.js
/**
 * CordovaiOSKeyboardPlugin
 * ======================================================
 * Apache Cordova Plugin for iOS
 * Written by Marc Weiner (mhweiner234@gmail.com)
 * https://github.com/mhweiner/CordovaiOSKeyboardPlugin
 * License: MIT
 */


'use strict';

var exec = cordova.require('cordova/exec');

var success = function () {
    // do nothing, succeed quietly
};

var is_open = false;
var keyboard_height = 0;

/**
 * Called by Obj-C code
 * @param height
 */
function keyboardWillShow(height){
    keyboard_height = height;
    var elem = document.getElementByTagName('body');
    var event = new Event('keyboardWillShow');
    elem.dispatchEvent(event);
}

/**
 * Called by Obj-C code
 */
function keyboardWillHide(){
    $('body').trigger('keyboardWillHide');
    var elem = document.getElementByTagName('body');
    var event = new Event('keyboardWillHide');
    elem.dispatchEvent(event);
}

/**
 * Called by Obj-C code
 */
function keyboardDidShow(){
    is_open = true;
    var elem = document.getElementByTagName('body');
    var event = new Event('keyboardDidShow');
    elem.dispatchEvent(event);
}

/**
 * Called by Obj-C code
 */
function keyboardDidHide(){
    is_open = false;
    var elem = document.getElementByTagName('body');
    var event = new Event('keyboardDidHide');
    elem.dispatchEvent(event);
}

function isOpen(){
    return is_open;
}

function getHeight(){
    return keyboard_height;
}

module.exports = {
    keyboardWillShow: keyboardWillShow,
    keyboardWillHide: keyboardWillHide,
    keyboardDidShow: keyboardDidShow,
    keyboardDidHide: keyboardDidHide,
    isOpen: isOpen,
    getHeight: getHeight
};

File: src/ios/KeyboardPlugin.m
/**
 * CordovaiOSKeyboardPlugin
 * ======================================================
 * Apache Cordova Plugin for iOS
 * Written by Marc Weiner (mhweiner234@gmail.com)
 * https://github.com/mhweiner/CordovaiOSKeyboardPlugin
 * License: MIT
 */

#import "KeyboardPlugin.h"
#import <QuartzCore/QuartzCore.h>

@implementation KeyboardPlugin


- (void) pluginInitialize
{

    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillShow:)
                                                 name:UIKeyboardWillShowNotification
                                               object:nil];

    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardWillHide:)
                                                 name:UIKeyboardWillHideNotification
                                               object:nil];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardDidShow:)
                                                 name:UIKeyboardDidShowNotification
                                               object:nil];
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(keyboardDidHide:)
                                                 name:UIKeyboardDidHideNotification
                                               object:nil];
    
    NSLog(@"Keyboard plugin initialized.");

}

- (void)keyboardWillShow:(NSNotification *)notif {

    NSDictionary *userInfo = notif.userInfo;

    // Get keyboard size.

    NSValue *endFrameValue = userInfo[UIKeyboardFrameEndUserInfoKey];
    CGRect keyboardEndFrame = [self.viewController.view convertRect:endFrameValue.CGRectValue fromView:nil];

    // Get height of keyboard for JS callback
    
    CGFloat height = keyboardEndFrame.size.height;

    // Call JS event handler
    
    NSString *cmd = [NSString stringWithFormat:@"Keyboard.keyboardWillShow(%f);", height];
    [self.commandDelegate evalJs:cmd];

}

- (void)keyboardWillHide:(NSNotification *)notif {
    
    [self.commandDelegate evalJs:@"Keyboard.keyboardWillHide();"];
    
}

- (void)keyboardDidShow:(NSNotification *)notif {
    
    [self.commandDelegate evalJs:@"Keyboard.keyboardDidShow();"];
    
}

- (void)keyboardDidHide:(NSNotification *)notif {

    [self.commandDelegate evalJs:@"Keyboard.keyboardDidHide();"];
    
}

@end